samcli/lib/sync/infra_sync_executor.py

```diff
@@ -0,0 +1,404 @@
+
+        Parameters
+        ----------
+        packaged_template_path : str
+            The template location of the current template packaged
+        built_template_path : str
+            The template location of the current template built
+        stack_name : str
+            The CloudFormation stack name that the template is deployed to
+        nested_prefix: Optional[str]
+            The nested stack stack name tree for child stack resources
+
+        Returns
+        -------
+        bool
+            Returns True if no template changes from last deployment
+            Returns False if there are template differences
+        """
+        current_template = self.get_template(packaged_template_path)
+        current_built_template = self.get_template(built_template_path)
+
+        if not current_template or not current_built_template:
+            LOG.debug("Cannot obtain a working current template for template path")
+            return False
+
+        try:
+            last_deployed_template_str = self._cfn_client.get_template(
+                StackName=stack_name, TemplateStage="Original"
+            ).get("TemplateBody", "")
+        except ClientError as ex:
+            LOG.debug("Stack with name %s does not exist on CloudFormation", stack_name, exc_info=ex)
+            return False
+
+        last_deployed_template_dict = yaml_parse(last_deployed_template_str)
+
+        sanitized_current_template = copy.deepcopy(current_template)
+        sanitized_last_template = copy.deepcopy(last_deployed_template_dict)
+
+        sanitized_resources = self._sanitize_template(
+            sanitized_current_template, built_template_dict=current_built_template
+        )
+        self._sanitize_template(sanitized_last_template, linked_resources=sanitized_resources)
+
+        if sanitized_last_template != sanitized_current_template:
+            LOG.debug("The current template is different from the last deployed version, we will not skip infra sync")
+            return False
+
+        # The recursive template check for Nested stacks
+        for resource_logical_id in current_template.get("Resources", {}):
+            resource_dict = current_template.get("Resources", {}).get(resource_logical_id, {})
+            resource_type = resource_dict.get("Type")
+
+            if resource_type in CODE_SYNCABLE_RESOURCES:
+                last_resource_dict = last_deployed_template_dict.get("Resources", {}).get(resource_logical_id, {})
+                resource_resolved_id = nested_prefix + resource_logical_id if nested_prefix else resource_logical_id
+
+                if resource_type == AWS_LAMBDA_FUNCTION:
+                    if not resource_dict.get("Properties", {}).get("Code", None) == last_resource_dict.get(
+                        "Properties", {}
+                    ).get("Code", None):
+                        self._code_sync_resources.add(resource_resolved_id)
```

Comment:
Shouldn't we check that resource have a local path to add it into `self._code_sync_resources`?

Reply:
Local path is already ensured in the removal logic, if not local, we will not sanitize at all, therefore returning value None for resource_resolved_id